Test/Unit/Controller/Cart/EmailTest.php

```diff
@@ -0,0 +1,182 @@
+        $this->request->method('getParam')->willReturn('');
+        $exception = new LocalizedException(__('No email received.'));
+        $this->bugsnag->expects($this->once())
+                    ->method('notifyException')
+                    ->with($this->equalTo($exception));
+        $this->currentMock->execute();
+    }
+
+    /**
+     * This doesn't really seem to test invalid emails as the validateEmail method
+     * needs to be mocked. Generally just making sure that if validateEmail comes
+     * back false we error handle properly
+     * @covers Email::execute
+     */
+
+    public function testInvalidEmail()
+    {
+        $this->checkoutSession->method('getQuote')->willReturn($this->quote);
+        $this->request->method('getParam')->willReturn('invalidemail');
+        $this->cartHelper->method('validateEmail')->willReturn(false);
+        $exception = new LocalizedException(__('Invalid email: %1', 'invalidemail'));
+        $this->bugsnag->expects($this->once())
+                    ->method('notifyException')
+                    ->with($this->equalTo($exception));
+        $this->currentMock->execute();
+    }
+
+    public function testOtherException()
+    {
+        $this->bugsnag->expects($this->once())->method('notifyException');
+        $this->currentMock->execute();
+    }
+
+    /**
+     * @test
+     * @covers Email::execute
+     */
+//
+    public function testExecuteHappyPath()
+    {
+        $this->checkoutSession->method('getQuote')->willReturn($this->quote);
+        $this->cartHelper->method('validateEmail')->willReturn(true);
+        $this->request->method('getParam')->willReturn('example@email.com');
+        $this->checkoutSession->expects($this->once())->method('getQuote');
+        $this->cartHelper->expects($this->once())->method('quoteResourceSave');
+        $this->currentMock->execute();
+    }
+
+    private function initRequiredMocks()
+    {
+        //objects needed for constructor
+        $this->context = $this->createMock(Context::class);
+        $this->checkoutSession = $this->createMock(CheckoutSession::class);
+        $this->customerSession = $this->createMock(CustomerSession::class);
+
+        $this->bugsnag = $this->createMock(Bugsnag::class);
+
+        $this->cartHelper = $this->createMock(CartHelper::class);
+
+        //objects needed for method
+        $this->quote = $this->createMock(Quote::class);
+        $this->quote->method('getId')->willReturn('1234');
+
+        $this->request = $this->createMock(RequestInterface::class);
+
+        //methods
+//        $this->checkoutSession->method('getQuote')->willReturn($this->quote);
+        $this->cartHelper->method('quoteResourceSave')->willReturn(null);
+        $this->context->method('getRequest')->willReturn($this->request);
+    }
+
+    private function initCurrentMock()
+    {
+        $this->currentMock = $this->getMockBuilder(Email::class)
```

Comment:
Can we use the object manager to initialize the object?

https://github.com/BoltApp/bolt-magento2/blob/9586db85834ee0e20fb2eb9c0c555bfea055c616/Test/Unit/Model/FeatureSwitchRepositoryTest.php#L44

If not why do we need a mock based object to test the current method? Won't it mock out things we should be testing?

Reply:
I'm assuming `enableProxyingToOriginalMethods()` alleviates some issues but curious

Reply:
I was following what was done in `Model/PaymentTest.php` for all the test structure. `enableProxyingToOriginalMethods()` does make sure the method is called correctly.

Reply:
I feel like this method is prone to skipping the actual methods. @daisy1754 @matt-thomason what's the preferred way here? I sort of prefer new Object manager way but that's with no deep thought into the issue ( only bcos the other seems like it could lead to trouble)

Reply:
enableProxyingToOriginalMethods should be OK